### PR TITLE
use UDUNITS seconds_per_year

### PR DESCRIPTION
**Pull request by MarcoGorelli**
As discussed on Slack, let's use UDUNITS in seconds_per_year

If users have a calendar other than Gregorian, then conversion needs to happen before they use pyleoclim

### CODE REVIEW

**Comment by CommonClimate** [comment]
Hi @MarcoGorelli , thanks for completing this. In terms of review, it is a little hard to tell what was done with the forced pushed changes.  Can you summarize in what way the datetimes are "more precise"? Is it literally because UDUNITS has more decimal points in its definition of the number of days per year than 365.25? I'm asking because it looks like you also made quite a few changes in tsbase.py, and would like to understand what they are doing and why (at a high level).

**Comment by MarcoGorelli** [comment]
Hey

originally I was interpreting `20.1` as meaning `20 years plus 0.1 times the number of seconds in the next year`, whereas on Slack we discussed that it should mean `20.1 * SECONDS_PER_YEARS`. So I amended the logic to do that - it's much simpler like that it was before, which is nice

The other changes are just adding extra tests, docstrings, and amending the existing ones

**Comment by CommonClimate** [comment]
OK, I'll merge, then.
